util/log,*: enforce literal strings in non-formatting log calls

knz opened the pull request:
This patch changes the logging interface and the `fmtsafe` linter to enforce that uses of the non-formatting log functions (`log.Info`, `log.Warning` etc) always uses a literal string.

This makes it possible to assume that the argument to those functions
is PII-free and can be collected in PII-free logs and telemetry.

knz committed:
util/log: change the non-formatting loggers to accept just a string

The previous signature of the non-formatting functions was as follows:

```go
   log.Info(ctx context.Context, args ...interface{})
```

This patch changes it to be just:

```go
   log.Info(ctx context.Context, msg string)
```

The variadic version is still available via the `...f()` alternative.

This change is motivated by a desire to separate PII-laden and
PII-free logging bits. This change creates an incentive for
users to call the `...f()` variant where the format string is
considered PII-free.

Release note: None

knz committed:
*: ensure the non-formatting logs only use literal strings

This patch changes the `fmtsafe` linter to enforce that caller of the
non-formatting log functions (`log.Info`, `log.Warning` etc) always
uses a literal string.

This makes it possible to assume that the argument to those functions
is PII-free and can be collected in PII-free logs and telemetry.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48048)
<!-- Reviewable:end -->

knz committed:
util/log: make `Shout()`/`Shoutf()` work like `Info()`/`Infof()`

This introduces the sorely-missed formatting variant `log.Shoutf`,
and introduces the requirement that the first string argument
to both `log.Shout` and `log.Shoutf` be a string literal and
thus PII-free suitable for reporting.

Release note: None

irfansharif reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 54 of 54 files at r1, 94 of 94 files at r2, 28 of 28 files at r3, 13 of 13 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48048#-:-M60oIHXDzGm16GtTo8I:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @knz)

___
*[pkg/kv/kvserver/raft_log_queue.go, line 600 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/48048#-M60lsbi2coP6XQRJAXk:-M60lsbi2coP6XQRJAXl:bviqgor) ([raw file](https://github.com/cockroachdb/cockroach/blob/dbd3e5633cdaf37d6e95bb0fb941b6e1bc717856/pkg/kv/kvserver/raft_log_queue.go#L600)):*
> ```Go
> 	if decision.ShouldTruncate() {
> 		if n := decision.NumNewRaftSnapshots(); log.V(1) || n > 0 && rlq.logSnapshots.ShouldProcess(timeutil.Now()) {
> 			// TODO(knz,tbg): check how much of this can be included in log.Safe().
> ```

All of it, same comment as https://github.com/cockroachdb/cockroach/pull/48040#pullrequestreview-402026998.
___
*[pkg/sql/logictest/logic.go, line 2730 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/48048#-M60kiku60sjTrWhoySK:-M60kiku60sjTrWhoySL:bwf8tf) ([raw file](https://github.com/cockroachdb/cockroach/blob/fabd65e75df1685435ee5cede3c3178c17f2508a/pkg/sql/logictest/logic.go#L2730)):*
> ```Go
> // "FAIL" marker when -show-sql is set. It also registers the error to the
> // failure counter.
> func (t *logicTest) Error(args ...interface{}) {
> ```

Perhaps add a TODO here to have `logicTest.Error{f}` and variants have the same kind of API as we're enforcing elsewhere now.


<!-- Sent from Reviewable.io -->

knz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48048)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @irfansharif)

___
*[pkg/kv/kvserver/raft_log_queue.go, line 600 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/48048#-M60lsbi2coP6XQRJAXk:-M6BNdGvDTBy0jNOFrtg:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dbd3e5633cdaf37d6e95bb0fb941b6e1bc717856/pkg/kv/kvserver/raft_log_queue.go#L600)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

All of it, same comment as https://github.com/cockroachdb/cockroach/pull/48040#pullrequestreview-402026998.
</blockquote></details>

Done.
___
*[pkg/sql/logictest/logic.go, line 2730 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/48048#-M60kiku60sjTrWhoySK:-M6BNld79HA4CeJmw7JG:b-nziimh) ([raw file](https://github.com/cockroachdb/cockroach/blob/fabd65e75df1685435ee5cede3c3178c17f2508a/pkg/sql/logictest/logic.go#L2730)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

Perhaps add a TODO here to have `logicTest.Error{f}` and variants have the same kind of API as we're enforcing elsewhere now.
</blockquote></details>

Nah, this does not matter: the output from this never makes it to telemetry nor does it ever contain customer-sensitive data.


<!-- Sent from Reviewable.io -->

knz commented:
bors r=irfansharif
